DOCS-4587 distributed file polling (#401)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/distributed-file-polling.adoc

= Distributed File Polling
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: file polling, cluster

*_Enterprise Edition_*

> Some endpoints, such as the File Endpoint or FTP Endpoint, regularly poll directories and read certain files as they are created in the directories polled. These files may reside on a remote file system, including file systems of nodes belonging to a Mule High Availability (HA) Cluster.

Reviewer: ```suggestion
Some connectors, such as the File connector or FTP connector, regularly poll directories and read certain files as they are created in the directories polled. These files may reside on a remote file system, including file systems of nodes belonging to a Mule High Availability (HA) Cluster.
```

Author: Updated

> Some connectors, such as the File connector or FTP connector, regularly poll directories and read certain files as they are created in the directories polled. These files may reside on a remote file system, including file systems of nodes belonging to a Mule High Availability (HA) Cluster.

Reviewer 2: avoid "may". Not sure what to recommend. I assume they sometimes (but not always) reside on a remote file system? If sometimes, I'll check to see what our suggested replacement is. I'd normally recommend "can" but need to see if that rec should change.

Also,  isn't it only the On New File operation in the File or FTP connector that actually polls regularly in Mule 4?

Author: @aiannucci Can this be confirmed?

Reviewer 2: You can say that the connectors poll. Just more precise that it's the operation.

Author: Updated

> Up to Mule 3.3.x, all file poll requests sent to a cluster were channeled through a single node in that cluster. This ensured file consistency by avoiding the possibility of file system sync issues, but had the disadvantage of introducing a bottleneck, reducing the performance of the cluster.

Reviewer: I think we should remove the comparison with Mule 3.3.x since it no longer makes sense

Author: Updated

> In the current version file polling for clusters' distributed file systems makes it possible to poll files in all cluster nodes. Enabled by default, this feature is used by the following endpoints:

Reviewer: ```suggestion
In the current version, file polling for clusters' distributed file systems makes it possible to poll files in all cluster nodes. Enabled by default, this feature is used by the following connectors:
```

Reviewer: Explain the `primaryNodeOnly` feature of said connectors

Author: Hi Ale, Could you provide details on the `primaryNodeOnly` feature for the connectors? 
I was able to find this information https://github.com/mulesoft/mulesoft-docs/pull/2280/files regarding this field.
What you would like to be added for the connectors?

Author: Hi Ale, Could you provide details on the primaryNodeOnly feature for the connectors?
I was able to find this information https://github.com/mulesoft/mulesoft-docs/pull/2280/files regarding this field.
What you would like to be added for the connectors?

Reviewer: It's exactly that. Just mention that connectors can be explicitly configured to only poll from the primary node, thus ignoring the default setting set by the runtime. This is only available in 4.x

Author: Updated.

* xref:connectors::ftp/ftp-connector.adoc[FTP Connector]
* xref:connectors::sftp/sftp-connector.adoc[SFTP Connector]

> You can configure connectors to only poll from the primary node, `@PrimaryNodeOnly`. Therefore, ignoring the default setting set by the Mule runtime engine. This is only available in Mule 4.x

Reviewer 2: Suggested fix to fix sentence (which is a sentence fragment):  "You can configure ... `@PrimaryNodeOnly`, which ignores the default ..."

"This" needs to modify something. Perhaps "This feature is only ..."  That said, if these are Mule 4 docs, it might not be necessary to mention that it's only available in Mule 4. Your call. If you think it will help 3.x customers who are moving to Mule 4, leave it in.

Author: Updated.

> * xref:connectors::file/file-connector.adoc[File Endpoint]

Reviewer: ```suggestion
* xref:connectors::file/file-connector.adoc[File Connector]
```

Author: Updated

> * xref:connectors::ftp/ftp-connector.adoc[FTP Endpoint]

Reviewer: ```suggestion
* xref:connectors::ftp/ftp-connector.adoc[FTP Connector]
```

Author: Updated

> * xref:connectors::sftp/sftp-connector.adoc[SFTP Endpoint]

Reviewer: ```suggestion
* xref:connectors::sftp/sftp-connector.adoc[SFTP Connector]
```

Author: Updated
